hail/src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -186,7 +186,12 @@ object PruneDeadFields {
           case ta: TStreamable =>
             ta.copyStreamable(unifySeq(ta.elementType, children.map(_.asInstanceOf[TStreamable].elementType)))
           case _ =>
-            assert(children.forall(_.asInstanceOf[Type].isOfType(t)))
+
+            if (!children.forall(_.asInstanceOf[Type].isOfType(t))) {
```

Comment:
Not a big deal but could just do the filter once and check if badChildren is empty

Reply:
that'll allocate a new IndexedSeq in the non-failure case, which I'd prefer to avoid.

Reply:
ah gotcha, makes sense